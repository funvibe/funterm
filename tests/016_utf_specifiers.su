// Test UTF specifiers in bitstring patterns
lua.data = "Hello"

match lua.data {
    << h/utf8, e/utf8, l1/utf8, l2/utf8, o/utf8 >> -> {
        lua.print("Matched UTF-8 codepoints:", h, e, l1, l2, o)
    },
    _ -> lua.print("No match")
}

lua.data_cyrllic = "Привет"

match lua.data_cyrllic {
    << p/utf8, r/utf8, i/utf8, v/utf8, e/utf8, t/utf8 >> -> {
        lua.print("Matched cyrillic UTF-8 codepoints:", p, r, i, v, e, t)
    },
    _ -> lua.print("No match")
}