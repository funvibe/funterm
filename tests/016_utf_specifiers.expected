Matched UTF-8 codepoints: 72 101 108 108 111
Matched cyrillic UTF-8 codepoints: 1055 1088 1080 1074 1077 1090
